### PR TITLE
Stabilize thread::is_finished

### LINKED ISSUE #90470: Tracking Issue for JoinHandle::is_finished
Feature gate: `#![feature(thread_is_running)]`

### Public API

```rust
impl<T> JoinHandle<T> {
    pub fn is_finshed(&self) -> bool;
}
```

### Steps / History

- [x] Implementation: https://github.com/rust-lang/rust/pull/90439
- [ ] ~~First Final comment period (FCP)~~: https://github.com/rust-lang/rust/issues/90470#issuecomment-1040742419
- [x] Rename it to `is_finished`: https://github.com/rust-lang/rust/pull/94549
- [x] Final comment period (FCP): https://github.com/rust-lang/rust/issues/90470#issuecomment-1062728858
- [x] Stabilization PR: https://github.com/rust-lang/rust/pull/95130

### Unresolved Questions

- [x] Do we also want to provide a way to access the result of the thread without having to .join() it? See https://github.com/rust-lang/rust/pull/90439#issuecomment-956256273
  - Answer: Maybe, but not a blocker for this feature.
- [x] Do we also want this in the form of a Future? See https://github.com/rust-lang/rust/pull/90439#issuecomment-955746106
  - Answer: Maybe, but not a blocker for this feature.
- [x] Name it `is_finished` instead?
  - Answer: Yes.

### PR DESCRIPTION
**Pull request by workingjubilee**
Closes #90470.

r? @m-ou-se

### CODE REVIEW

File: library/std/src/thread/scoped.rs [hunk 1]
```diff
@@ -316,7 +316,7 @@ impl<'scope, T> ScopedJoinHandle<'scope, T> {
     ///
     /// This function does not block. To block while waiting on the thread to finish,
     /// use [`join`][Self::join].
-    #[unstable(feature = "thread_is_running", issue = "90470")]
+    #[stable(feature = "thread_is_running", since = "1.61.0")]
```
Comment (m-ou-se) [inline library/std/src/thread/scoped.rs:319]: Scoped threads aren't stable yet, so this should remain unstable like the rest of ScopedJoinHandle.
```suggestion
```

**Comment by Dylan-DPC** [comment]
potential failure of [rollup](https://github.com/rust-lang/rust/pull/95472#issuecomment-1083327221)

**Comment by m-ou-se** [comment]
"warning: static variable `statik` should have an upper case name"

That seems unrelated. https://github.com/rust-lang/rust/pull/94806 maybe?

@bors r+

**Pull request merged**